gnovm/pkg/gnolang/machine.go

```diff
@@ -154,13 +148,16 @@ var (
 // and prevent objects that were not taken from
 // the pool, to call Release
 func (m *Machine) Release() {
-	// copy()
 	// here we zero in the values for the next user
 	m.NumOps = 0
 	m.NumValues = 0
-	// this is the fastest way to zero-in a slice in Go
-	copy(m.Ops, opZeroed[:])
-	copy(m.Values, valueZeroed[:])
+
+	m.Ops = make([]Op, VMSliceSize)
+	m.Values = make([]TypedValue, VMSliceSize)
+	m.Exprs = nil
+	m.Stmts = nil
+	m.Blocks = nil
+	m.Frames = nil
```

Comment:
What if instead we did this?

```suggestion
	ops, values := m.Ops[:VMSliceSize:VMSliceSize], m.Values[:VMSliceSize:VMSliceSize]
	copy(ops, opZeroed[:])
	copy(values, valueZeroed[:])
	*m = Machine{Ops: ops, Values: values} 
```

This way we zero-in all other values, while also resetting correctly without reallocating Ops and Values. The remaining part of both slices (exceeding `VMSliceSize`) is outside of the cap, and can be safely collected by the gc...

Calling `make` here makes the whole machinePool shenanigans more useless than it currently is.

Consequently, could update `NewMachineWithOptions` by making it get the machine from the pool and then only changing the fields it needs to change.

(I think @piux2 missed that part of the code, which is probably the whole reason why the gnovm kept working nonetheless. Although not zeroing in everything else in this fn probably lead to memory leaks).